module-bsp/board/rt1051/puretx/bsp/keyboard/keyboard.cpp

```diff
@@ -67,101 +77,140 @@ namespace bsp::keyboard
-                                          .defLogic = 0,
-                                          .pin      = static_cast<uint32_t>(BoardDefinitions::KEYBOARD_IRQ_PIN)});
+    void clearAllIRQs()
+    {
+        std::uint8_t clearIRQs = 0xFF;
+        i2cAddr.subAddress     = REG_INT_STAT;
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&clearIRQs), 1);
+    }
 
+    void configureRegisters()
+    {
         std::uint32_t reg = 0;
-        status_t error    = 0;
-
-        gpio->WritePin(static_cast<std::uint32_t>(BoardDefinitions::KEYBOARD_RESET_PIN), 0);
-        vTaskDelay(1);
-        gpio->WritePin(static_cast<std::uint32_t>(BoardDefinitions::KEYBOARD_RESET_PIN), 1);
-        vTaskDelay(10);
-
         /* Assemble a mask for row and column registers */
         reg = ~(std::uint32_t(~0) << TCA8418_ROWS_COUNT);
         reg += (~(std::uint32_t(~0) << TCA8418_COL_COUNT)) << 8;
 
         /* Set registers to keypad mode */
         i2cAddr.subAddress = REG_KP_GPIO1;
-        i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
         reg                = reg >> 8;
         i2cAddr.subAddress = REG_KP_GPIO2;
-        i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
         reg                = reg >> 16;
         i2cAddr.subAddress = REG_KP_GPIO3;
-        i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
 
         /* Enable column debouncing */
         i2cAddr.subAddress = REG_DEBOUNCE_DIS1;
-        i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
         reg                = reg >> 8;
         i2cAddr.subAddress = REG_DEBOUNCE_DIS2;
-        i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
         reg                = reg >> 16;
         i2cAddr.subAddress = REG_DEBOUNCE_DIS3;
-        i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
-
-        if (error != kStatus_Success) {
-            return error;
-        }
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
 
         reg = CFG_INT_CFG | CFG_OVR_FLOW_IEN | CFG_KE_IEN;
         // Enable interrupts
         i2cAddr.subAddress = REG_CFG;
-        i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
+        write(i2cAddr, reinterpret_cast<std::uint8_t *>(&reg), 1);
 
         // Get key pressed/released count
         std::uint8_t val   = 0;
         i2cAddr.subAddress = REG_KEY_LCK_EC;
-        i2c->Read(i2cAddr, reinterpret_cast<std::uint8_t *>(&val), 1);
+        read(i2cAddr, reinterpret_cast<std::uint8_t *>(&val), 1);
```

Comment:
we don't check the return codes on purpose?

Reply:
More or less, yes. After powerup, the keyboard chip is reset and then we assume that communication works fine. I've never seen any issue with the keyboard while initialization. 
I discussed with @Lefucjusz and @awulkiew about initialization and as far as we don't know the exact issue we don't reset the chip.

Reply:
alright, but I think that we should handle it in the future ;)

Reply:
Definitely, I agree :)